### ISSUE TITLE
en-dash and i-circumflex translation error

### ISSUE THREAD

**Issue by a-lhn** [retorquere/zotero-better-bibtex#2796]
### Debug log ID

EUCKUCRJ-refs-apse/6.7.162-6 , 3N376R9P-refs-apse/6.7.162-6

### What happened?

Behaviour of unicode to latex commands suddenly changed - was previously working consistently. Some entries now convert circumflex i to \^{\i}, instead of \^{i} which causes an error. In addition, previously endashes were converted to {\textendash}  which displayed correctly, but now as converted to -- which does not.

**Comment by retorquere** [comment]
Hi, do you have an MWE where `\^{\i}` causes an error? And how does `--` render differently than `\textendash`? I thought these two were equivalent?

**Comment by a-lhn** [comment]
Hi here's the MWE reproducing the circumflex error. I will double check the endash behaviour, I may have been too hasty (but it may have occurred in mathmode or something like that)
[MSW Circumflex Error.zip](https://github.com/retorquere/zotero-better-bibtex/files/14396027/MSW.Circumflex.Error.zip)

**Comment by retorquere** [comment]
For future reference -- you can make single-file MWEs using `filecontents`, but this compiles without error for me (although I see now LaTeX is smart enough to make `\^i` and `\^\i` the same these days):

```
\documentclass{article}

\usepackage[backend=biber, style=alphabetic]{biblatex}

\begin{filecontents}{\jobname.bib}
@inproceedings{Cervantes-cota,
  title = {Cosmology today---{{A}} brief review},
  author = {{Cervantes-Cota}, Jorge L. and Smoot, George},
  year = {2011},
  eprint = {1107.1789v2},
  pages = {28--52},
  doi = {10.1063/1.3647524},
  archiveprefix = {arxiv},
  keywords = {review}
}

@article{Lemaitre1931,
  title = {A {{Homogeneous Universe}} of {{Constant Mass}} and {{Increasing Radius}} accounting for the {{Radial Velocity}} of {{Extra-galactic Nebulae}}},
  author = {Lema\^{\i}tre, A. G.},
  year = {1931},
  month = mar,
  journal = {Monthly Notices of the Royal Astronomical Society},
  volume = {91},
  number = {5},
  pages = {483--490},
  doi = {10.1093/mnras/91.5.483}
}
\end{filecontents}
\addbibresource{\jobname.bib}

\begin{document}


\section{MSW Circumflex Error}
Here is a citation \cite{Lemaitre1931} which causes an error.
\printbibliography[heading=bibintoc]
\end{document}
```

**Comment by retorquere** [comment]
the en-dash should not be showing up in math mode, but it doesn't for me.

**Comment by retorquere** [comment]
I'll change the i-circumflex back to what it was; it seems latex already makes the `i` dotless these days for these cases.

**Comment by retorquere** [comment]
I've changed `\^\i` back, but I can't replicate compilation errors with `--`.

**Comment by damiencourousse** [comment]
> For future reference -- you can make single-file MWEs using `filecontents`, but this compiles without error for me (although I see now LaTeX is smart enough to make `\^i` and `\^\i` the same these days):
> 
> ```
> \documentclass{article}
> 
> \usepackage[backend=biber, style=alphabetic]{biblatex}
> 
> \begin{filecontents}{\jobname.bib}
> @inproceedings{Cervantes-cota,
>   title = {Cosmology today---{{A}} brief review},
>   author = {{Cervantes-Cota}, Jorge L. and Smoot, George},
>   year = {2011},
>   eprint = {1107.1789v2},
>   pages = {28--52},
>   doi = {10.1063/1.3647524},
>   archiveprefix = {arxiv},
>   keywords = {review}
> }
> 
> @article{Lemaitre1931,
>   title = {A {{Homogeneous Universe}} of {{Constant Mass}} and {{Increasing Radius}} accounting for the {{Radial Velocity}} of {{Extra-galactic Nebulae}}},
>   author = {Lema\^{\i}tre, A. G.},
>   year = {1931},
>   month = mar,
>   journal = {Monthly Notices of the Royal Astronomical Society},
>   volume = {91},
>   number = {5},
>   pages = {483--490},
>   doi = {10.1093/mnras/91.5.483}
> }
> \end{filecontents}
> \addbibresource{\jobname.bib}
> 
> \begin{document}
> 
> 
> \section{MSW Circumflex Error}
> Here is a citation \cite{Lemaitre1931} which causes an error.
> \printbibliography[heading=bibintoc]
> \end{document}
> ```



MWE:
```
$ cat test2.tex
\documentclass{article}

\usepackage[backend=biber, style=alphabetic]{biblatex}

\begin{filecontents}{\jobname.bib}
@inproceedings{Cervantes-cota,
  title = {Cosmology today---{{A}} brief review},
  author = {{Cervantes-Cota}, Jorge L. and Smoot, George},
  year = {2011},
  eprint = {1107.1789v2},
  pages = {28--52},
  doi = {10.1063/1.3647524},
  archiveprefix = {arxiv},
  keywords = {review}
}

@article{Lemaitre1931,
  title = {A {{Homogeneous Universe}} of {{Constant Mass}} and {{Increasing Radius}} accounting for the {{Radial Velocity}} of {{Extra-galactic Nebulae}}},
  author = {Lema\^{\i}tre, A. G.},
  year = {1931},
  month = mar,
  journal = {Monthly Notices of the Royal Astronomical Society},
  volume = {91},
  number = {5},
  pages = {483--490},
  doi = {10.1093/mnras/91.5.483}
}
\end{filecontents}
\addbibresource{\jobname.bib}

\begin{document}


\section{MSW Circumflex Error}
Here is a citation \cite{Lemaitre1931} which causes an error.
\printbibliography[heading=bibintoc]
\end{document}

```


build log:
```
$ rm test2.bib
$ latexmk -pdf -bibtex test2 -c
Rc files read:
  /etc/LatexMk
Latexmk: This is Latexmk, John Collins, 7 Jan. 2023. Version 4.79.
Latexmk: Doing main (small) clean up for 'test2.tex'

$ latexmk -pdf -bibtex test2
Rc files read:
  /etc/LatexMk
Latexmk: This is Latexmk, John Collins, 7 Jan. 2023. Version 4.79.
No existing .aux file, so I'll make a simple one, and require run of *latex.
Latexmk: applying rule 'pdflatex'...
Rule 'pdflatex':  Reasons for rerun
Category 'other':
  Rerun of 'pdflatex' forced or previously required

------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'pdflatex  -recorder  "test2.tex"'
------------
This is pdfTeX, Version 3.141592653-2.6-1.40.24 (TeX Live 2022/Debian) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./test2.tex
LaTeX2e <2022-11-01> patch level 1
L3 programming layer <2023-01-16>
(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2022/07/02 v1.4n Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/size10.clo))
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/biblatex.sty
(/usr/share/texlive/texmf-dist/tex/generic/pdftexcmds/pdftexcmds.sty
(/usr/share/texlive/texmf-dist/tex/generic/infwarerr/infwarerr.sty)
(/usr/share/texlive/texmf-dist/tex/generic/iftex/iftex.sty)
(/usr/share/texlive/texmf-dist/tex/generic/ltxcmds/ltxcmds.sty))
(/usr/share/texlive/texmf-dist/tex/latex/etoolbox/etoolbox.sty)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texlive/texmf-dist/tex/latex/kvoptions/kvoptions.sty
(/usr/share/texlive/texmf-dist/tex/latex/kvsetkeys/kvsetkeys.sty))
(/usr/share/texlive/texmf-dist/tex/latex/logreq/logreq.sty
(/usr/share/texlive/texmf-dist/tex/latex/logreq/logreq.def))
(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texlive/texmf-dist/tex/latex/url/url.sty)
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/blx-dm.def)
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/blx-compat.def)
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/biblatex.def)
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/bbx/alphabetic.bbx
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/bbx/standard.bbx))
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/cbx/alphabetic.cbx)
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/biblatex.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/l3kernel/expl3.sty
(/usr/share/texlive/texmf-dist/tex/latex/l3backend/l3backend-pdftex.def))
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/blx-case-expl3.sty
(/usr/share/texlive/texmf-dist/tex/latex/l3packages/xparse/xparse.sty)))

LaTeX Info: Writing file `./test2.bib'.


(./test2.aux) (/usr/share/texlive/texmf-dist/tex/latex/biblatex/lbx/english.lbx
)
No file test2.bbl.

LaTeX Warning: Citation 'Lemaitre1931' on page 1 undefined on input line 35.


LaTeX Warning: Empty bibliography on input line 36.

[1{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}] (./test2.aux)

LaTeX Warning: There were undefined references.


Package biblatex Warning: Please (re)run Biber on the file:
(biblatex)                test2
(biblatex)                and rerun LaTeX afterwards.

 )</usr/share/texlive/texmf-dist/fonts/type1/public/amsfonts/cm/cmbx10.pfb></us
r/share/texlive/texmf-dist/fonts/type1/public/amsfonts/cm/cmbx12.pfb></usr/shar
e/texlive/texmf-dist/fonts/type1/public/amsfonts/cm/cmr10.pfb>
Output written on test2.pdf (1 page, 36751 bytes).
Transcript written on test2.log.
Latexmk: Getting log file 'test2.log'
Latexmk: Examining 'test2.fls'
Latexmk: Examining 'test2.log'
Latexmk: Missing bbl file 'test2.bbl' in following:
 No file test2.bbl.
Latexmk: Log file says output to 'test2.pdf'
Latexmk: Bibliography file(s) form .bcf file:
  ./test2.bib
Latexmk: applying rule 'biber test2'...
Rule 'biber test2':  Reasons for rerun
Category 'other':
  Rerun of 'biber test2' forced or previously required

------------
Run number 1 of rule 'biber test2'
------------
------------
Running 'biber  "test2.bcf"'
------------
INFO - This is Biber 2.18
INFO - Logfile is 'test2.blg'
INFO - Reading 'test2.bcf'
INFO - Found 1 citekeys in bib section 0
INFO - Processing section 0
INFO - Looking for bibtex file 'test2.bib' for section 0
INFO - LaTeX decoding ...
INFO - Found BibTeX data source 'test2.bib'
INFO - Overriding locale 'en-US' defaults 'variable = shifted' with 'variable = non-ignorable'
INFO - Overriding locale 'en-US' defaults 'normalization = NFD' with 'normalization = prenormalized'
INFO - Sorting list 'anyt/global//global/global' of type 'entry' with template 'anyt' and locale 'en-US'
INFO - No sort tailoring available for locale 'en-US'
INFO - Writing 'test2.bbl' with encoding 'UTF-8'
INFO - Output to test2.bbl
Latexmk: Found biber source file(s) [test2.bcf test2.bib]
Latexmk: applying rule 'pdflatex'...
Rule 'pdflatex':  Reasons for rerun
Changed files or newly in use/created:
  test2.aux
  test2.bbl
  test2.run.xml

------------
Run number 2 of rule 'pdflatex'
------------
------------
Running 'pdflatex  -recorder  "test2.tex"'
------------
This is pdfTeX, Version 3.141592653-2.6-1.40.24 (TeX Live 2022/Debian) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./test2.tex
LaTeX2e <2022-11-01> patch level 1
L3 programming layer <2023-01-16>
(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2022/07/02 v1.4n Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/size10.clo))
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/biblatex.sty
(/usr/share/texlive/texmf-dist/tex/generic/pdftexcmds/pdftexcmds.sty
(/usr/share/texlive/texmf-dist/tex/generic/infwarerr/infwarerr.sty)
(/usr/share/texlive/texmf-dist/tex/generic/iftex/iftex.sty)
(/usr/share/texlive/texmf-dist/tex/generic/ltxcmds/ltxcmds.sty))
(/usr/share/texlive/texmf-dist/tex/latex/etoolbox/etoolbox.sty)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texlive/texmf-dist/tex/latex/kvoptions/kvoptions.sty
(/usr/share/texlive/texmf-dist/tex/latex/kvsetkeys/kvsetkeys.sty))
(/usr/share/texlive/texmf-dist/tex/latex/logreq/logreq.sty
(/usr/share/texlive/texmf-dist/tex/latex/logreq/logreq.def))
(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texlive/texmf-dist/tex/latex/url/url.sty)
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/blx-dm.def)
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/blx-compat.def)
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/biblatex.def)
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/bbx/alphabetic.bbx
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/bbx/standard.bbx))
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/cbx/alphabetic.cbx)
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/biblatex.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/l3kernel/expl3.sty
(/usr/share/texlive/texmf-dist/tex/latex/l3backend/l3backend-pdftex.def))
(/usr/share/texlive/texmf-dist/tex/latex/biblatex/blx-case-expl3.sty
(/usr/share/texlive/texmf-dist/tex/latex/l3packages/xparse/xparse.sty)))

LaTeX Info: File `test2.bib' already exists on the system.
            Not generating it from this source.

(./test2.aux) (/usr/share/texlive/texmf-dist/tex/latex/biblatex/lbx/english.lbx
) (./test2.bbl)

! LaTeX Error: Unicode character ̂ (U+0302)
               not set up for use with LaTeX.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.36 \printbibliography[heading=bibintoc]
                                         
? x
No pages of output.
Transcript written on test2.log.
Latexmk: Getting log file 'test2.log'
Latexmk: Examining 'test2.fls'
Latexmk: Examining 'test2.log'
Latexmk: Found input bbl file 'test2.bbl'
Latexmk: Log file says no output from latex
Latexmk: For rule 'pdflatex', no output was made
Latexmk: Bibliography file(s) form .bcf file:
  ./test2.bib
Latexmk: Summary of warnings from last run of *latex:
  =====Latex reported missing or unavailable character(s).
=====See log file for details.
Latexmk: ====List of undefined refs and citations:
  ! LaTeX Error: Unicode character ̂ (U+0302)
Latexmk: Errors, so I did not complete making targets
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 1
      Refer to 'test2.log' and/or above output for details

Latexmk: If appropriate, the -f option can be used to get latexmk
  to try to force complete processing.

```

**Comment by retorquere** [comment]
> author = {Lema\^{\i}tre, A. G.},

That doesn't look like output from 5882 or 5861 though. Can you try again with 5883?

**Comment by retorquere** [comment]
I'm waiting on this issue to put out a new release.

**Comment by retorquere** [comment]
@a-lhn if you're not going to respond I'm going to assume your issue is resolved and close when @damiencourousse confirms.

**Comment by a-lhn** [comment]
Sorry, I'm in the middle of editing an important document and I don't want to mess with installing the test build immediately. If you would like me to test it could it please wait until next week?

**Comment by retorquere** [comment]
From the issue template:

> Please be available in the days following your report for follow-up questions and testing.

I know people have stuff to get done, which is why I generally try to get a fix out within a day. I clear important stuff of my own out of the way to make this happen, and my time is relatively scarce. And other people might be hit with the bug (like @damiencourousse), and I do not want to have them wait.

If @damiencourousse confirms I'm going to release the fix. If it turns out that it does not resolve the error, please open a new issue.

**Comment by retorquere** [comment]
I'm going to consider this fixed
